### DISCUSSION TITLE
Automatic adjustment of coordinates

### DISCUSSION THREAD

**Discussion by psads-git** [cetz-package/cetz#678] [Q&A] [answered]
Consider the following code. My question is: If one moves the second circle (`circ2`) vertically, one needs to _manually_ adjust the following parameter of function line `(rel: (0,2))`: Can one have a more automatic way that adjusts that coordinate? Maybe resolving `circ2` coordinates? But how? 

```
#import "@preview/cetz:0.2.2"

#cetz.canvas({
  import cetz.draw: *

  set-style(circle: (radius: 0.8em))
  circle((0,0), name: "circ1")
  circle((2,2), name: "circ2")

  line("circ1.east", (rel: (0.6,0)), (rel: (0,2)), "circ2.west", 
     mark: (start: ">", fill: black))
})
```

![image](https://github.com/user-attachments/assets/d64ed51f-2d87-4f79-acc2-a0b57d9e3142)

**Comment by johannes-wolf** [comment, marked as answer]
I guess you want something dynamic like the following:
```typ
#import "@preview/cetz:0.2.2"
#set page(width: auto, height: auto)

#cetz.canvas({
  import cetz.draw: *

  set-style(circle: (radius: 0.8em))
  circle((0,0), name: "circ1")
  circle((2,2), name: "circ2")

  
  line("circ1",                             /* Intersection of a straight line between circ1.center and the next coordinate */
       ((), "-|", ("circ1", 50%, "circ2")), /* Move only horizontal to (circ1.x + circ2.x)/2 */
       ((), "|-", "circ2"),                 /* Move vertically to circ2.y */
       "circ2",                             /* Same as 1. but for circ2 */
      mark: (start: ">", fill: black))
})
```

With `"circ2"` at `(2,2)`:
![grafik](https://github.com/user-attachments/assets/32f54f79-b816-4eba-bbb1-a66ffb19aba7)

With `"circ2"` at `(-3,1)`:
![grafik](https://github.com/user-attachments/assets/b75b75cd-741f-416f-ae1d-ff17e0a2b06a)

**Comment by psads-git** [comment]
That is a perfect solution, @johannes-wolf! Thank you very much indeed!
